### ISSUE TITLE
[Feature Request] Modify mp3 id3 tags for easy import to mp3 players (eg. Plex)

### ISSUE THREAD

**Issue by captainnapalm** [akhilrex/podgrab#23]
I'd love to see a way to modify the mp3 id3 tags within podgrab.

I'm going to use [Self-Hosted](https://feeds.fireside.fm/selfhosted/rss)'s rss as an example

My ideal scenario is:
- Title: `<item><title>`
- Artist: `<channel><title>`
- Album Artist: `<item><author>`
- Album: `<channel><title>`
- Comment: `<item><description>`

In theory, for each podcast this should create a single artist within Plex, with a single album, and all episodes within that one album. I haven't quite figured out how best to do track number/year/etc.

**Comment by Code-Slave** [comment]
yea would love this. trying to put this into any player app is a nightmare

**Comment by akhilrex** [comment]
I did look into this but a lot of podcasts already have ID3 tags built into their files while some do not. I will revisit this soon.

**Comment by captainnapalm** [comment]
that's all a guy can ask for who doesn't have the skills to actually contribute. main issue is that, while the id3 tags are there in most cases, they're not done consistently.

**Comment by Code-Slave** [comment]
Yea the tags are usually a mess. they will change mid seasons the album etc for no reason. Be nice to force a set of tags and have it set.

**Comment by akhilrex** [comment]
Update: I had started work on this when the original request came in but the testing results were very random so that branch was abandoned for more deliverable work. I will revisit this and try to close it. I am currently building functionality to set ID3 tags if not present.

**Comment by Code-Slave** [comment]
awesome, thanks

**Comment by Extarys** [comment]
I tried the latest version of podgrab (first time user, great job btw) and use a Music library inside Jellyfin. It works but indeed metadata need some love.

I copied `folder.png` to `cover.png` so I can see the image of the podcast inside Jellyfin. The only podcast I tried so far doesn't have ID3 tags so I'll copy them from podgrab UI manually.

For the curious this is how it looks "raw":
![image](https://user-images.githubusercontent.com/11581433/130335605-b93794c8-1c3d-4965-abbb-d00e3ebb2106.png)

How I'm trying to edit each episode:
![image](https://user-images.githubusercontent.com/11581433/130335726-e2c3faa1-6ade-41ed-8a8f-583fe6105c49.png)

How it (starting to) looks like after edit:
![image](https://user-images.githubusercontent.com/11581433/130335769-4c34ab25-cc92-445e-bde5-ee88ff412805.png)

I would like to have the podcast image attached to the mp3 but I would need to create an album for each episode and download the image form podgrab to that album folder so Jellyfin can display it :/, which I don't think is worth the time.

For now I'm glad I found a podcast grabber for my Jellyfin :joy:

**Comment by HStep20** [comment]
Hope this is still in the pipeline. Been a bit since an update on it, and the branch is a bit outdated as well.

**Comment by billdeys** [comment]
This would be a game changer for me. I've always wanted to get my podcasts into plex, so I can rely on my phone less but the ID3 tags in podcasts are terrible. Being able to a) use my NAS as a podcatcher and 2) use plex to listen would be a dream come true!

**Comment by tapatianbeast** [comment]
https://www.upi.com/Top_News/US/2022/04/12/plex-remove-podcast-web-shows-friday-not-worth-our-time/1031649775957/

**Comment by captainnapalm** [comment]
@tapatianbeast Yes, Plex is removing their podcast feature, which is separate from building your own podcast library using podgrab.

**Comment by ghenry22** [comment]
One way this could be handled predictably would be to be able to set overrides for key fields for each podcast.  Then you always replace those tags regardless what already exists whenever an episode is downloaded.

Key for sorting / filtering when adding to music management apps like plex, subsonic, jellyfin etc would be:

album
artist
albumartist
genre: Podcast

This would be a great first step with limited complexity and would allow podcasts to be neatly grouped and browsed at a basic level.

Potential extension:

Populate the year tag with the release date for the episode.
Populate the comment tag with the brief episde description.
Populate the track tag sequentially based on on release date. (this provides basic chronological ordering within non-podcast specific apps). podgrab already does this with the track numbers on the file names so it's just a matter of writing the same to the file tag.

**Comment by HStep20** [comment]
For those still interested in this, I recommend checking out AudioBookshelf. Its an open source Audiobook/Podcast server which handles downloading, storing metadata for, and playing podcasts (and audiobooks!) to IOS and Android clients, as well as any web client. 

Ive moved on from podgrab + plex, since ABS does everything I need for archiving and streaming all in one :)
